### DISCUSSION TITLE
need a name for the device values

### DISCUSSION THREAD

**Discussion by proddy** [emsesp/EMS-ESP32#119] [Ideas]
On terminology we have

*  **device** = A device type like Thermostat, Boiler, Solar Module. (system is also a device in the code, which is awkward)
*  **device value** = Like 'curflowtemp' in the device Boiler

but what I've always been missing is a name for the actual origin of the device value. Today we use the word "device value" interchangeably between the name and the actual value.

Any ideas what we should call it? For example 'device parameter or 'variable' or 'entity' ?

**Comment by proddy** [comment]
For example, the class `DeviceValue` in `emsdevice.h` which has all the information about a device's value (name, type, uom etc..) should perhaps have a different name. Now we have a "device value's value" which could be 21.5 

@MichaelDvP  ?

**Comment by MichaelDvP** [comment]
I don't see this critical, if you really want to rename my suggestion is "device item"

**Comment by proddy** [comment]
I don't intend to change any code, that would be a nightmare! It's just a word to be used in comments and discussions when we talk about something like 'boiltemp'. It's an **_item_** of the device 'boiler' then?

**Reply by MichaelDvP** [reply]
[In the doc](https://emsesp.github.io/docs/#/MQTT) you call it `key`: "The key's and their description:" 
That's also ok: device-key-shortname, device-key-value, etc.
